Add new feature about screen

ZineeEddine opened the pull request:
Hi, @davidepanidev   Here is a preview of the new About Screen feature

We try to apply what you have asked us. @beastboym

The logo in the center as well as the title "About", we chose to make cards with the logo and the title and information.

Different sections to realize (Source Code, Contact, Support)

which open the different links

Do you want us to leave the arrow for the return?

We don't think it has any use, because that's what the navigation buttons below are for.

Here are 2 screenshots and 1 video to illustrate.

https://user-images.githubusercontent.com/92603736/220655500-30f936a7-9a42-4b6d-9685-d9af096dd9a4.mp4

<img src="https://user-images.githubusercontent.com/92603736/220655487-c1f9d197-46a6-4b90-bc45-dfc537359687.png"  width="440" height="780">
<img src="https://user-images.githubusercontent.com/92603736/220655496-f26443ac-475b-49ba-b6ba-7c8de6e222a9.png"  width="440" height="780">

davidepanidev commented:
Hi @ZineeEddine!

in order to accept the pull request it needs to be 100% compliant with the requirements. Here are the points that are not compliant with what was specified in #12:

-  no new fourth About tab is expected. You just have to submit the Screen which will not be visible until the next release when **we** will add the Settings tab.
- no About title is expected
- the items of a same section must be grouped toghether and not placed on separate cards. The items must share the same rounded corner background and be separated by a white line separator. It must follow **exactly** the same structure of CoinDetailScreen.
- The Donate item must not contain the address but the link to the support section of the README (https://github.com/CoinTrend#support)


Below are the UI improvement that also must be followed to make the UI consistent with the rest of the App:

- The CoinTrend title should be a little larger in terms of text size than the sections' titles.
- Each section must be properly spaced from the above one. E.g. add a spacer of 16dp between each section.
- The title of each item, like "Mail" should follow **exactly** the same style (size, style, color) of the crypto name in the crypto cards.
- The subtitle of each item, like "cointrend.info@gmail.com" should follow **exactly** the same style (size, style, color) of the crypto symbol (e.g. BTC) in the crypto cards.
- The right padding of each of the items is greater than the padding on the left. It must be the **same** and **exactly the same one** of the Market Data section of CoinDetailScreen.


Below are the code improvements:

- The outer Column item below the TopAppBar that I see on your file seems to me useless. The outer item should be the LazyList.
- All the items inside the LazyList must be placed each one in **seperate** _item {}_.


Moreover I expect to see exactly the following changes in your commit:

- A new file called **AboutScreen** placed in a new packed called **about** inside the existing **ui** package;
- One or more added drawables in the **presentation** module

**No other changes to any other file will be accepted**, unless properly motivated.

Please, take your time to submit quality code that is consistent with the style of the rest of the App. When ready we will do another iteration like this one.

ZineeEddine commented:
@davidepanidev  apologies, @ZineeEddine  @beastboym we'll start over and the interface and removing all other changes and we will come back to you for more details.

ZineeEddine commented:
Hi @davidepanidev,
we @ZineeEddine @beastboym have made the following changes,
Can you tell us, if we are working well, if it meets your expectations?

We kept the navigation, just to see the AboutScreen screen at compile time (interface changes), we will remove all the changes later, and we will keep the AboutScreen.kt file

Section elements, for example Github Issues, should they be clickable and open the link, or just display the link?

Should the background of the elements be gray like cypto or dark like market data?

Here is an illustration of the changes (we tried to respect the instructions as much as possible: padding, sections, size, divider)


<br>
<p float="left">
  <img src="https://user-images.githubusercontent.com/92603736/221032985-5a545abf-b2cf-48e4-8c1a-1e9b48f8ad0e.png" width="23%" />
  <img src="https://user-images.githubusercontent.com/92603736/221032986-bf87445a-0958-410f-ab7c-1b16adcfd5da.png" width="23%" />
</p>

What do you think of this new improvement?

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+        modifier = Modifier,
+        topBar = {
+            TopAppBar(
+                title = {IconButton(onClick = {
+                    System.out.println("Go Back")
+                }) {
+                    Icon(
+                        painter = painterResource(id = R.drawable.ic_arrow_back_ios),
+                        contentDescription = "Return to previous screen",
+                        modifier = Modifier.padding(start = 12.dp)
+                    )
+                }
```
This is not the correct way. Just copy lines from 58 to 67 of CoinDetailScreen

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+            )
+        }
+    ) { padding ->
+        Column(
```
This outer Column seems to me useless. Try to remove it and move the attributes to the LazyColumn below

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+                        modifier = Modifier
+                            .padding(horizontal = 8.dp)
+                            .background(
+                                color = StocksDarkBackgroundTranslucent,
```
Here let's try again with the same background color of the crypto cards as you did the previous time. It seemed better actually

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+        Image(
+            modifier = Modifier
+                .requiredHeight(120.dp)
+                .padding(top = 2.dp, start = 115.dp),
```
The padding 'start' must be removed

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+                .padding(top = 2.dp, start = 115.dp),
+            contentScale = ContentScale.Fit,
+            painter = painterResource(id = R.drawable.ic_launcher),
+            contentDescription = null,
```
Please, add a content description. Just  "CoinTrend logo icon"

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+            contentDescription = null,
+        )
+        Text(
+            modifier = Modifier.padding(top = 2.dp, start = 115.dp),
```
The padding 'start' must be removed

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+        Text(
+            modifier = Modifier.padding(top = 2.dp, start = 115.dp),
+            text = "CoinTrend",
+            fontSize = 38.sp,
```
Dont't specify directly the test size. Use the style attribute below

davidepanidev replied:
Here you have not removed the 'fontSize = 38.sp' yet

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+            text = "CoinTrend",
+            fontSize = 38.sp,
+            color = MaterialTheme.colorScheme.onBackground,
+            style = MaterialTheme.typography.bodySmall,
```
Here try other styles which have a larger text size than the 'MaterialTheme.typography.titleLarge' one

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+
+        Image(
+            modifier = Modifier
+                .size(size = 30.dp),
```
Apply also this
`.clip(shape = MaterialTheme.shapes.medium)`

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+                color = MaterialTheme.colorScheme.onPrimaryContainer,
+                maxLines = 1
+            )
+            Spacer(modifier = Modifier.size(4.dp))
```
Reduce to 2.dp

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+                text = info,
+                style = MaterialTheme.typography.bodyMedium,
+                color = MaterialTheme.colorScheme.onSecondaryContainer,
+                modifier = Modifier.padding(start = 0.dp, end = 4.dp, top = 1.dp, bottom = 1.dp),
```
Try to remove those padding

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+        modifier = Modifier
+            .padding(8.dp)
+            .fillMaxWidth(),
+        horizontalArrangement = Arrangement.spacedBy(8.dp),
```
Try with 16.dp

davidepanidev replied:
Here change "Arrangement.spacedBy(8.dp)" with "Arrangement.spacedBy(16.dp)"

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+    ) {
+    Row(
+        modifier = Modifier
+            .padding(8.dp)
```
Try with 16.dp, at least for the horizontal one

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis
+            )
+            Spacer(modifier = Modifier.size(4.dp))
```
Remove this Spacer

davidepanidev replied:
This Spacer has to be removed yet

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+            modifier = Modifier
+                .padding(horizontal = 8.dp)
+                .alpha(.2f),
+            color = Color(0xffffffff)
```
Use Color.White

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/coindetail/CoinDetailScreen.kt:
```diff
@@ -66,17 +67,6 @@ fun CoinDetailScreen(
                     }
                 },
                 actions = {
```
This file should not be changed

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/coindetail/CoinDetailScreen.kt:
```diff
@@ -66,17 +67,6 @@ fun CoinDetailScreen(
-
                     IconButton(
                         onClick = {
                             viewModel.onFavouriteButtonClick()
```
This file should not be changed

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+                                shape = MaterialTheme.shapes.large
+                            ),
+                    ) {
+                        SectionInfoItemTest(name = "Donate Bitcoin", info = "https://github.com/CoinTrend/CoinTrend#bitcoin", image = R.drawable.ic_donate, showDivider = true)
```
The correct URL is 'https://github.com/CoinTrend#support' and the title is just 'Donate'

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+                                shape = MaterialTheme.shapes.large
+                            ),
+                    ) {
+                        SectionInfoItemTest(name = "Github", info = "https://github.com/CoinTrend/CoinTrend", image = R.drawable.ic_github, showDivider = false)
```
GitHub, not Github

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+                item {
+                    CoinTrend()
+                }
+
```
Add a Spacer of 16.dp here

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+                            ),
+                    ) {
+                        SectionInfoItemTest(name = "Github", info = "https://github.com/CoinTrend/CoinTrend", image = R.drawable.ic_github, showDivider = false)
+                    }
```
In this column let's add also an item
"Changelog" -> "https://github.com/CoinTrend/CoinTrend/releases"

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,236 @@
+                                shape = MaterialTheme.shapes.large
+                            ),
+                    ) {
+                        SectionInfoItemTest(name = "Email", info = "cointrend.info@gmail.com", image = R.drawable.ic_gmail, showDivider = true)
```
Try to find a better email icon. Maybe there is one in the Material library that I used across the App like `Icons.Default.TrendingUp`

davidepanidev reviewed (changes requested):
Ok great! I made here a detailed review line-by-line so that you know what has to be changed to fine tune the UI. Thanks

ZineeEddine commented:
Hi, @davidepanidev 
I applied the following changes and here is the result obtained on the screen:

I removed the "about" item from the navigation bar but also from the MainActivity.kt, Screen.kt and Enum.kt file.

<p float="left">
  <img src="https://user-images.githubusercontent.com/92603736/221438987-a061d6c6-30c1-4f09-a00b-8e1c722cc276.png" width="43%" />
  <img src="https://user-images.githubusercontent.com/92603736/221438990-afef78d7-f488-46ba-b7e5-d3e60fe86139.png" width="43%" />
</p>

Unless I am mistaken.    
Here are the changes to be made that can be seen on the screenshot above.   

- :white_check_mark:    reverted 7.2.2 to 7.3.1
- :white_check_mark:    Supress the focusManager
- :white_check_mark:    copying lines from 58 to 67 of CoinDetailScreen
- :white_check_mark:    Deletion and moved the attributes to the LazyColumn below
- :white_check_mark:    Added the same background color of the crypto cards
- :white_check_mark:    Deletion padding 'start'
- :white_check_mark:    Added a content description "CoinTrend logo icon" for the logo
- :white_check_mark:    Deletion padding 'start'
- :white_check_mark:    Dont't specify directly the fontSize. Use the style attribute below
- :white_check_mark:    Here try other styles which have a larger text size than the 'MaterialTheme.typography.titleLarge' one
- :white_check_mark:    Apply also this .clip(shape = MaterialTheme.shapes.medium)
- :white_check_mark:    reduce 2dp
- :white_check_mark:    Try to remove those padding (after text=info,style,color 2nd composable )
- :white_check_mark:    Try with 16.dp  :point_down:
- :white_check_mark:    Try with 16.dp, at least for the horizontal one
- :white_check_mark:    Remove this Spacer
- :white_check_mark:    Use of white color with Color.White 
- :white_check_mark:    Added correct URL 'https://github.com/CoinTrend#support' and 'Donate' title
- :white_check_mark:    Correction GitHub, not Github
- :white_check_mark:    Added Spacer 16dp after CoinTrend()
- :white_check_mark:    Added item "Changelog" -> "https://github.com/CoinTrend/CoinTrend/releases"
- :white_check_mark:    Added new mail icon

We hope that the modifications correspond to your expectations, what do you think ?

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/coindetail/CoinDetailScreen.kt:
```diff
@@ -57,7 +57,8 @@ fun CoinDetailScreen(
                 title = {},
                 navigationIcon = {
-                    IconButton(onClick = { navController.pop() }) {
+                    IconButton(onClick = {
```
All changes to this file must be reverted

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/about/AboutScreen.kt:
```diff
@@ -0,0 +1,210 @@
+    ) {
+    Row(
+        modifier = Modifier
+            .padding(horizontal =  16.dp)
```
Here add padding ", vertical = 8.dp"

davidepanidev reviewed (changes requested):
There are some changes that are still to be made. Check them out and submit the final pull request. Please, make sure to check the Android Studio options inside the commit dialog before committing: "Cleanup code", "Rearrange code", "Optimize imports", "Analyze code" so that the code follows the Kotlin formatting standards.

ZineeEddine commented:
@davidepanidev 
We @beastboym @ZineeEddine  made the following changes, here is the result of the screen below:

We have deleted in the AboutScreen.kt file, the imports never used, we have carried out a kotlin formatting, and clean code.
The SectionItemInfoTest composable is called now SectionItemInfoAbout for a better understanding of the code.

We hope you like it and meet your expectations.

<p float="left">
<img src="https://user-images.githubusercontent.com/92603736/221699100-a9ab9f8d-87fa-4ee8-b210-d3c7af2007f0.png" width="43%" />
  <img src="https://user-images.githubusercontent.com/92603736/221699097-88354cc7-b3fa-497f-9f1b-13cb7fae0be3.png" width="43%" />
</p>

davidepanidev commented on presentation/src/main/java/com/cointrend/presentation/ui/coindetail/CoinDetailScreen.kt:
```diff
@@ -66,17 +66,6 @@ fun CoinDetailScreen(
                     }
                 },
                 actions = {
-                    /*
-                    IconButton(onClick = {}) {
-                        Icon(
-                            imageVector = Icons.Filled.Notifications,
-                            contentDescription = "Localized description",
-                            tint = Favourite
-                        )
-                    }
-
-                     */
-
```
There is still this change in CoinDetailScreen. Please, revert it.

beastboym replied:
it's good we have applied the changes

davidepanidev reviewed (changes requested):
Ok there is one last change to be reverted in CoinDetailScreen. Then, I think we are ready.

davidepanidev commented:
@ZineeEddine @beastboym welcome aboard! Thanks for contributing 🙏

ZineeEddine commented:
@davidepanidev  Thank you for allowing us to work on the project.

It was my first time using JetPack Compose, and I enjoyed it.

Thank you, and good luck with your future updates.

beastboym commented:
> @davidepanidev Thank you for allowing us to work on the project.
> 
> It was my first time using JetPack Compose, and I enjoyed it.
> 
> Thank you, and good luck with your future updates.

Same for me, good luck 😄
